factor/auth/classes/factor.php

```diff
@@ -48,28 +48,25 @@ public function setup_user_factor($data) {
      * {@inheritDoc}
      */
     public function get_all_user_factors() {
+        global $DB, $USER;
+        $records = $DB->get_records('tool_mfa', array('userid' => $USER->id, 'factor' => $this->name));
 
-        $factor = (object) array(
-            'id' => 1,
-            'name' => $this->name,
-            'devicename' => '-',
-            'timecreated' => '-',
-            'createdfromip' => '-',
-            'lastverified' => '-',
-            'revoked' => '-'
-        );
-
-        return [$factor];
-    }
+        if (!empty($records)) {
+            return $records;
+        }
 
-    /**
-     * Auth Factor implementation.
-     * Factor cannot be revoked, so no extra functionality is needed.
-     *
-     * {@inheritDoc}
-     */
-    public function get_active_user_factors() {
-        return $this->get_all_user_factors();
+        // Null records returned, build new record.
+        $record = array(
+            'userid' => $USER->id,
+            'authtype' => $USER->auth,
```

Comment:
Why do you need to store auth type? This can change fairly easily and it would be simpler to not have to track it here